AudioLibrary.WMME/WaveOutBuffer.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using AudioLibrary.WMME.Native;
+
+namespace AudioLibrary.WMME
+{
+    internal class WaveOutBuffer : IDisposable
+    {
+        private readonly object _lockObj = new object();
+
+        public WaveOutBuffer NextBuffer;
+
+        private AutoResetEvent m_PlayEvent = new AutoResetEvent(false);
+        private IntPtr m_WaveOut;
+
+        private WAVEHDR m_Header;
+        private byte[] m_HeaderData;
+        private GCHandle m_HeaderHandle;
+        private GCHandle m_HeaderDataHandle;
+
+        private bool m_Playing;
+
+        internal static void WaveOutProc(IntPtr hdrvr, int uMsg, int dwUser, ref WAVEHDR wavhdr, int dwParam2)
+        {
+            if (uMsg == WaveNative.MM_WOM_DONE)
+            {
+                try
+                {
+                    GCHandle h = (GCHandle)wavhdr.dwUser;
+                    WaveOutBuffer buf = (WaveOutBuffer)h.Target;
+                    buf.OnCompleted();
+                }
+                catch
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Handle the exception or explain in a comment why it can be ignored. ([csharpsquid:S2486](https://sonarcloud.io/organizations/alex-yevtushenko3820/rules#rule_key=csharpsquid%3AS2486))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=IPP&pullRequest=7&issues=AWfG1-gu8DroMTOoScWw&open=AWfG1-gu8DroMTOoScWw)

[ISSUE_KEY]: AWfG1-gu8DroMTOoScWw